radix-common/src/data/manifest/model/manifest_expression.rs

```diff
@@ -20,25 +20,72 @@ pub enum ManifestExpression {
 
 #[cfg_attr(feature = "fuzzing", derive(Arbitrary))]
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum BucketBatch {
+pub enum ManifestBucketBatch {
     ManifestBuckets(Vec<ManifestBucket>),
     EntireWorktop,
 }
 
-impl BucketBatch {
+pub trait ResolvableBucketBatch {
+    fn resolve(self, resolver: &impl NamedManifestBucketResolver) -> ManifestBucketBatch;
+}
+
+impl<B: ResolvableManifestBucket> ResolvableBucketBatch for Vec<B> {
+    fn resolve(self, resolver: &impl NamedManifestBucketResolver) -> ManifestBucketBatch {
+        let buckets = self.into_iter().map(|b| b.resolve(resolver)).collect();
+        ManifestBucketBatch::ManifestBuckets(buckets)
+    }
+}
+
+impl<B: ResolvableManifestBucket> ResolvableBucketBatch for BTreeSet<B> {
```

Comment:
We can provide a generic implementation for all types that implement `IntoIterator<Item = B>`. Example:

```rust
impl<T, B> ResolvableBucketBatch for T where T: IntoIterator<Item = B>, B: ExistingManifestBucket {
    fn consume_and_resolve(self, registrar: &ManifestNameRegistrar) -> BucketBatch {
        let buckets = self
            .into_iter()
            .map(|b| b.mark_consumed(registrar))
            .collect();
        BucketBatch::ManifestBuckets(buckets)
    }
}
```

Reply:
It clashes with the other generic implementation.